textdb/textdb-common/src/main/java/edu/uci/ics/textdb/common/field/SpanTuple.java

```diff
@@ -0,0 +1,39 @@
+package edu.uci.ics.textdb.common.field;
+
+import java.util.List;
+
+import edu.uci.ics.textdb.api.common.Attribute;
+import edu.uci.ics.textdb.api.common.IField;
+
+public class SpanTuple extends DataTuple{
+    
+    private IField spanField;
+    private Span span;
+    private String key;
+
+    public SpanTuple(List<Attribute> schema, IField[] fields, 

```

Comment:
Got it. Then we should take that Tuple inside, and recalculate the schema and fields. Because the SpanTuple is still an `ITuple`, if I call `SpanTuple.getSchema()` it should return an extended schema, and also the `.getFields()` related methods.